web/__tests__/pages/api/post/post.int.test.ts

```diff
@@ -145,4 +155,138 @@ describe("[API] Post - Integration Test", () => {
+        }
+      }
+      const postFilterSchema = {
+        type: [coveredPost.type],
+        breed: coveredPost.breed,
+        age: [coveredPost.age],
+        size: [coveredPost.size],
+        gender: [coveredPost.gender],
+        behavioral: coveredPost.behavioral,
+        goodWith: goodWithArray,
+      };
+      const filteredPosts = await caller.post.getFilteredPosts({
+        postFilterSchema: postFilterSchema,
+      });
+      expect(filteredPosts).not.toBeNull();
+      expect(filteredPosts.length).toBe(2);
+      const coveredPosts = await caller.post.getFilteredPosts({
+        postFilterSchema: postFilterSchema,
+        covered: true,
+      });
+      expect(coveredPosts).not.toBeNull();
+      expect(coveredPosts.length).toBe(1);
+      const uncoveredPosts = await caller.post.getFilteredPosts({
+        postFilterSchema: postFilterSchema,
+        covered: false,
+      });
+      expect(uncoveredPosts).not.toBeNull();
+      expect(uncoveredPosts.length).toBe(1);
+    });
+
+    test("random covered-uncovered", async () => {
+      let randomPost = createRandomPost();
+      const numPosts = Math.ceil(Math.random() * 10);
+      let numCovered = 0;
+      let numUncovered = 0;
+      let createPost;
+      for (let i = 0; i < numPosts; i++) {
+        const curr = Math.floor(Math.random() * 2);
+        if (curr == 0) {
+          randomPost.covered = false;
+          numUncovered++;
+        } else {
+          randomPost.covered = true;
+          numCovered++;
+        }
+        const id = new mongoose.Types.ObjectId();
+        createPost = await Post.create({
+          ...randomPost,
+          _id: id,
+          attachments,
+        });
+        expect(createPost).not.toBeNull();
+      }
+      let goodWithArray = [];
+      for (const goodWithKey in goodWithMap) {
+        if (createPost[goodWithKey] == "yes") {
+          goodWithArray.push(goodWithMap[goodWithKey]);
+        }
+      }
+      const postFilterSchema = {
+        type: [randomPost.type],
+        breed: randomPost.breed,
+        age: [randomPost.age],
+        size: [randomPost.size],
+        gender: [randomPost.gender],
+        behavioral: randomPost.behavioral,
+        goodWith: goodWithArray,
+      };
+      const allPosts = await caller.post.getFilteredPosts({
+        postFilterSchema: postFilterSchema,
+      });
+      expect(allPosts).not.toBeNull();
+      expect(allPosts.length).toBe(numPosts);
+      const coveredPosts = await caller.post.getFilteredPosts({
+        postFilterSchema: postFilterSchema,
+        covered: true,
+      });
+      expect(coveredPosts).not.toBeNull();
+      expect(coveredPosts.length).toBe(numCovered);
+      const uncoveredPosts = await caller.post.getFilteredPosts({
+        postFilterSchema: postFilterSchema,
+        covered: false,
+      });
+      expect(uncoveredPosts).not.toBeNull();
+      expect(uncoveredPosts.length).toBe(numUncovered);
```

Comment:
Use an array method to get the arrays of both covered and uncovered and use `expect(____).toMatchObject()`. This will match the two arrays and their nested values. We need this to correctly validate our implementation since the lengths alone are not enough (you should still keep the length checks and the null checks).

For the array method, you can look at how it's done in `__tests__/db/actions/User.unit.test.ts:62`.